src/inmanta/env.py

```diff
@@ -837,23 +834,24 @@ def _gen_content_requirements_file(cls, requirements_list: Sequence[str]) -> str
             except InvalidRequirement:
                 url = req_spec
 
-            if name not in modules:
-                modules[name] = {"name": name, "version": [], "markers": []}
+            requirement_id: str = name + "_" + str(marker) if marker else name
+            if requirement_id not in modules:
+                modules[requirement_id] = {"name": name, "version": [], "markers": []}
 
             if version is not None:
-                modules[name]["version"].extend(version)
+                modules[requirement_id]["version"].extend(version)
 
             if marker is not None:
-                modules[name]["markers"].append(marker)
-
+                modules[requirement_id]["markers"].append(marker)
             if url is not None:
-                modules[name]["url"] = url
+                modules[requirement_id]["url"] = url
 
             if extras is not None:
-                modules[name]["extras"] = extras
+                modules[requirement_id]["extras"] = extras
```

Comment:
This can also go wrong, because we set the list instead of appending to it. For example when the requirements.txt file contains:

```
dep[opt]
dep[otheropt]
```

It will results in:

```
dep[otheropt]
```